src/Lecture4.hs

```diff
@@ -131,8 +137,30 @@ errors. We will simply return an optional result here.
 🕯 HINT: Use the 'readMaybe' function from the 'Text.Read' module.
 -}
 
+trimSpaces :: String -> String
+trimSpaces = reverse . dropWhile isSpace . reverse . dropWhile isSpace
+
+splitOn :: Char -> String -> [String]
+splitOn _ [] = []
+splitOn sym str@(x:xs)
+  | x == sym = splitOn sym xs
+  | otherwise = clearedPart : splitOn sym other
+  where (part, other) = break (==sym) str
+        clearedPart = trimSpaces part
+
+parseCost :: String -> Maybe Int
+parseCost stringCost
+  | isNothing parsedValue = parsedValue
+  | fromJust parsedValue >= 0 = parsedValue
+  | otherwise = Nothing
+  where parsedValue = readMaybe stringCost
```

Comment:
This function is correct but it uses the `isNothing/fromJust` anti-pattern. `fromJust` is a partial function and you can avoid using it here by using pattern-matching:

```suggestion
parseCost :: String -> Maybe Int
parseCost stringCost = case parsedValue of
    Nothing -> Nothing
    Just x  -> if x >= 0 then Just x else Nothing
  where parsedValue = readMaybe stringCost
```

You can now notice that pattern-matching on `Maybe`, returning `Nothing` on `Nothing` and doing something with the `Just x` case is exactly the `Monad` instance for `Maybe` 🙂 So you can write your function in a slightly shorter way using monadic `Maybe` behaviour:

```suggestion
parseCost :: String -> Maybe Int
parseCost stringCost =
    readMaybe stringCost >>= \x -> if x >= 0 then Just x else Nothing
```